src/test/java/org/opensearch/security/dlic/rest/api/TenancyConfigApiTest.java

```diff
@@ -0,0 +1,93 @@
+package org.opensearch.security.dlic.rest.api;
+
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpStatus;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.test.helper.file.FileHelper;
+import org.opensearch.security.test.helper.rest.RestHelper;
+
+import static org.opensearch.security.OpenSearchSecurityPlugin.PLUGINS_PREFIX;
+
+public class TenancyConfigApiTest extends AbstractRestApiUnitTest {
+    private final String BASE_ENDPOINT;
+    private final String ENDPOINT;
+    protected String getEndpointPrefix() {
+        return PLUGINS_PREFIX;
+    }
+
+    public TenancyConfigApiTest(){
+        BASE_ENDPOINT = getEndpointPrefix();
+        ENDPOINT = getEndpointPrefix() + "/api/tenancyconfig";
+    }
+
+    @Test
+    public void testTenancyConfigAPIAccess() throws Exception {
+        Settings settings = Settings.builder().put(ConfigConstants.SECURITY_UNSUPPORTED_RESTAPI_ALLOW_SECURITYCONFIG_MODIFICATION, true).build();
+        setup(settings);
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        RestHelper.HttpResponse response = rh.executeGetRequest(ENDPOINT);
+        Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+
+        rh.sendAdminCertificate = false;
+        response = rh.executeGetRequest(ENDPOINT);
+        Assert.assertEquals(HttpStatus.SC_UNAUTHORIZED, response.getStatusCode());
+
+        rh.sendHTTPClientCredentials = true;
+        response = rh.executeGetRequest(ENDPOINT);
+        Assert.assertEquals(HttpStatus.SC_FORBIDDEN, response.getStatusCode());
+    }
+
+    @Test
+    public void testTenancyConfigAPIUpdate() throws Exception {
```

Comment:
these can be split into different tests for each property

Reply:
Thanks for the suggestion @shikharj05 . I will split these and update in next commit.

Reply:
I would suggest to keep it in one test, as the test sets up a cluster before executing the test and tears down after. This will impact on the performance of the tests.